node/src/actors/chain_manager/mod.rs

```diff
@@ -2008,6 +2011,20 @@ impl ChainManager {
         // Ice the invalid blocks' batch sender
         self.ice_peer(sender);
     }
+
+    /// Return the value of the version field for a block in this epoch
+    fn tapi_votes_mask(&self, epoch: Epoch) -> u32 {
+        let Tapi { oppose_wip0014 } = &self.tapi;
+
+        let mut v = 0;
+        // TODO: define epoch range for WIP0014 activation
+        let wip_0014_range = 500_000..=700_000;
```

Comment:
I can't see a need for activation ranges. Am I missing something?

Reply:
This allows us to reuse bits for a different WIP later on. For example, bit 0 will be used for the activation of WIP0014 between epochs 500_000 and 700_000, but it will also be used for the activation of WIP0066 between epochs 12_000_000 and 13_000_000.